proposals/lip-0042.md

```diff
@@ -165,12 +165,25 @@ The default reward of the block as a 64-bit unsigned integer.
 
 ### Block Processing
 
-#### After Block Execution
+#### Before Transactions Execution
+
+The following function initializes block generator account for `TOKEN_ID_REWARD`, if it has not been previously initialized. The `TOKEN_ID_REWARD` is the token ID of the token used for the reward system of the blockchain. 
+
+We refer to the `userSubstoreExists` and `initializeUserAccountForBlockGenerator` functions exposed by the [Token module][token-module].
+
+```python
+def beforeTransactionsExecute(b: Block) -> None:
+    (blockReward, reduction) = getBlockReward(b.header)
+    if blockReward > 0 and not Token.userSubstoreExists(b.header.generatorAddress, TOKEN_ID_REWARD):
+        Token.initializeUserAccountForBlockGenerator(b.header.generatorAddress, TOKEN_ID_REWARD)
+```
+
+#### After Transactions Execution
 
 The following function assigns block rewards after all transactions in the block are executed. Here we refer to the `mint` function exposed by the [Token module][token-module].
 
 ```python
-def afterTransactionsExecute(b: Block):
+def afterTransactionsExecute(b: Block) -> None:
     (blockReward, reduction) = getBlockReward(b.header)
```

Comment:
`(blockReward, reduction) = getBlockReward(b.header)` could be removed as you already calculated in the `beforeTransactionsExecute`.

Reply:
Resolving it since old changes are reverted.